disk/disk_linux.go

```diff
@@ -325,44 +325,63 @@ func parseFieldsOnMounts(lines []string, all bool, fs []string) []PartitionStat
 
 func parseFieldsOnMountinfo(ctx context.Context, lines []string, all bool, fs []string, filename string) ([]PartitionStat, error) {
 	ret := make([]PartitionStat, 0, len(lines))
+	seenDevIDs := make(map[string]string)
 
 	for _, line := range lines {
+		// See proc_pid_mountinfo(5) (proc(5) on EL)
 		// a line of 1/mountinfo has the following structure:
 		// 36  35  98:0 /mnt1 /mnt2 rw,noatime master:1 - ext3 /dev/root rw,errors=continue
 		// (1) (2) (3)   (4)   (5)      (6)      (7)   (8) (9)   (10)         (11)
 
 		// split the mountinfo line by the separator hyphen
-		parts := strings.Split(line, " - ")
+		parts := strings.SplitN(line, " - ", 2)
 		if len(parts) != 2 {
 			return nil, fmt.Errorf("found invalid mountinfo line in file %s: %s ", filename, line)
 		}
 
 		fields := strings.Fields(parts[0])
 		blockDeviceID := fields[2]
+		rootDir := fields[3]
 		mountPoint := fields[4]
 		mountOpts := strings.Split(fields[5], ",")
+		fields = strings.Fields(parts[1])
```

Comment:
Could you add length check?
```suggestion
		fields = strings.Fields(parts[1])
		if len(fields) < 6 {
			return nil, fmt.Errorf("invalid mountinfo line: %s", line)
		}

```

Reply:
Added length checks for both `fields` splits (the one highlighted here and the first split as well)